src/System.Private.ServiceModel/tests/Common/Scenarios/Endpoints.cs

```diff
@@ -1,49 +1,183 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
+using System.ServiceModel;
+using System.Security.Cryptography.X509Certificates;
+
 using Infrastructure.Common;
 
 public static partial class Endpoints
 {
+    private static X509Certificate2 rootCert;
+    private static X509Certificate2 clientCert;
+    private static object rootCertLock = new object();
+    private static object clientCertLock = new object();
+    //Install Root certificate, this is required for all https test
+    private static void EnsureRootCertificateInstalled()
+    {
+        lock (rootCertLock)
+        {
+            if (rootCert == null)
+            {
```

Comment:
Not blocking: note to capture for future thinking -- perhaps we should make the [ConditionalFact] for "Root_Certificate_Installed" be the one to call EnsureRootCertificateInstalled.  This would eliminate the need to try to keep both Endpoint.cs and the tests in sync regarding what they depend on.

---
In reply to: [59565735](https://github.com/dotnet/wcf/pull/1002#discussion_r59565735) [](ancestors = 59565735)